link/src/main/java/com/stripe/android/link/ui/wallet/WalletUiState.kt

```diff
@@ -0,0 +1,39 @@
+package com.stripe.android.link.ui.wallet
+
+import androidx.compose.runtime.Immutable
+import com.stripe.android.link.ui.PrimaryButtonState
+import com.stripe.android.model.ConsumerPaymentDetails
+
+@Immutable
+internal data class WalletUiState(
+    val supportedTypes: Set<String>,
+    val paymentDetailsList: List<ConsumerPaymentDetails.PaymentDetails>,
+    val selectedItem: ConsumerPaymentDetails.PaymentDetails?,
+    val isProcessing: Boolean,
+) {
+
+    val primaryButtonState: PrimaryButtonState
+        get() {
+            return if (isProcessing) {
+                PrimaryButtonState.Processing
+            } else {
+                PrimaryButtonState.Disabled
+            }
+        }
+
+    fun setProcessing(): WalletUiState {
+        return copy(
+            isProcessing = true,
+        )
+    }
+
+    fun updateWithResponse(
+        response: ConsumerPaymentDetails,
+    ): WalletUiState {
+        return copy(
+            paymentDetailsList = response.paymentDetails,
+            selectedItem = paymentDetailsList.firstOrNull { supportedTypes.contains(it.type) },
```

Comment:
should we filter out the payment details which are not in the supported types list? It looks like the payment details list could contain unsupported types

Reply:
We actually don't need to filter at all since we provided a set of supported types when making the request. I'll fix this.